text.cpp

```diff
@@ -91,7 +91,16 @@ class Text {
       }
 
       auto isMatchingSymbol = [&](const Symbol& symbol) -> bool {
-        return symbol.id == static_cast<uint32_t>(c);
+        /* c is char_t here, so ...
+          casting first to unsigned char is necessary, because if we
+          cast to int32_t directly, the machine sign extends, meaning
+          that the match would fail because of the type's sign.
+          we could also do uint32_t, since I think zengl supports utf-32
```

Comment:
We already do a 32 bit conversion here, so I'm not sure why you say "we could" - we already do.